src/main/docker/scripts/export-file-based-secrets.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+#
+# Copyright 2017-2021 Micro Focus or one of its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# A function for logging in the caf logging format.
+log() {
+    echo "$@" |& $(dirname "$0")/../scripts/caf-log-format.sh "export-file-based-secrets.sh"
```

Comment:
Because the `export-file-based-secrets.sh` script is sourced by `startup.sh`, this change was causing the logs echoed by `export-file-based-secrets.sh` to report that they were coming from `startup.sh`. 

I also found another bug in that I am sourcing this file (and thus this `log` function) which was replacing the `log` function in `startup.sh`.

I'm not sure what the best thing to do here is? I was thinking:

1) Rename the `log` function in this script to `caf_log`
2) Hardcode the `export-file-based-secrets.sh` name in the function as I had previously done

```
caf_log() {
    echo "$@" |& $(dirname "$0")/../scripts/caf-log-format.sh "export-file-based-secrets.sh"
}
```
3) As an extra measure, unset the `caf-log` function at the end of  `export-file-based-secrets.sh`, so that its not sourced into `startup.sh`

```
unset -f caf_log # Don't export the caf_log function when this script is sourced
```

This way the script names are correct in the logs:

```
[14:36:34.348Z #006.??? INFO  -            -   ] startup.sh: Running export-file-based-secrets.sh...
[14:36:34.357Z #00D.??? INFO  -            -   ] export-file-based-secrets.sh: Found environment variable ending with the _FILE suffix: ABC_PASSWORD_FILE=pw.txt, attempting to read the contents of pw.txt...
[14:36:34.364Z #016.??? INFO  -            -   ] export-file-based-secrets.sh: Successfully exported environment variable ABC_PASSWORD
[14:36:34.366Z #006.??? INFO  -            -   ] startup.sh: Running startup scripts...
```

Reply:
Tried this out here: https://github.com/CAFapi/opensuse-base-image/pull/34/commits/ce7a858d25df96680fda894eb67ca702a3a0e5b1 

Happy to change it though if you think there's a better way.

Reply:
It's fine.  There are a lot more subtleties here than I realized.